### PR TITLE
py-matplotlib: add v3.9.0

### PR DESCRIPTION
**Pull request by adamjstewart**
New release of matplotlib that uses meson!

### Before

  Stage: 0.93s.  Install: 31.12s.  Post-install: 0.70s.  Total: 32.91s

### After

  Stage: 0.95s.  Install: 12.47s.  Post-install: 0.65s.  Total: 14.58s

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-matplotlib/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-matplotlib/package.py]: @rgommers would you be interested in maintaining this one too? It seems very similar to numpy/scipy now that it also uses a meson build system.

Comment (rgommers) [reply var/spack/repos/builtin/packages/py-matplotlib/package.py]: Yes, I'd be happy to. I don't know as much about the Matplotlib, but there's a lot of commonality indeed so hopefully I can be helpful.

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
@adamjstewart Feel free to merge, if it seems ok to you. Not sure if you're waiting on a review from @rgommers

**Comment by adamjstewart** [comment]
Hmm, we seem to be building matplotlib 3.8.4 in CI instead. Any idea why?

**Comment by rgommers** [comment]
I don't see any obvious mistakes in the metadata, all LGTM. Prerequisites of 3.9.0 like `meson-python` and `ninja` are built, so no idea why 3.8.4 is selected.

**Comment by adamjstewart** [comment]
I'm able to reproduce this on the CLI.
```console
> spack spec --fresh py-matplotlib
```
chooses matplotlib 3.9, but
```console
> spack spec --fresh py-networkx
```
chooses matplotlib 3.8. Further investigation with `spack solve` reveals that choosing 3.9 results in more "version badness". This is because we depend on `py-ninja` instead of `ninja`, which forces us to use an older version of `ninja`. I think the easiest solution is to directly depend on `ninja`.

We've seen a lot of issues with `py-cmake` in the past. I wonder if we should remove `py-cmake` and `py-ninja` and only use the direct dependencies instead.

**Comment by alalazo** [comment]
> I wonder if we should remove py-cmake and py-ninja and only use the direct dependencies instead.

I would be for that. Aren't they pip packaging of the corresponding tool, for convenience of pip users?

**Comment by rgommers** [comment]
> Aren't they pip packaging of the corresponding tool, for convenience of pip users?

Yes they are. Agreed that depending on `ninja` is better.

**Comment by adamjstewart** [comment]
https://github.com/spack/spack/pull/44257

**Comment by adamjstewart** [comment]
Glad we tested newer matplotlib in CI because it's failing with the OneAPI compilers: https://gitlab.spack.io/spack/spack/-/pipelines/709134. Going to try re-enabling the hacks that were already there for this new version.

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Pull request merged**
